src/transformers/models/pix2struct/image_processing_pix2struct.py

```diff
@@ -43,11 +43,22 @@
 if is_torch_available():
     import torch
 
+    from transformers.pytorch_utils import is_torch_greater_or_equal_than_1_11
+else:
+    is_torch_greater_or_equal_than_1_11 = False
+
 
 logger = logging.get_logger(__name__)
 DEFAULT_FONT_PATH = "ybelkada/fonts"
 
 
+if is_torch_available() and not is_torch_greater_or_equal_than_1_11:
```

Comment:
Perhaps we'll want a way to specify versions using or with something similar to `requires_xxx` or `requires_backends(...)`. 

Generally, users don't read warnings, and so have an error closer to when the object is used - instantiation or method calling - could be useful.

Reply:
I agree, but partially. In some modeling file (`tapas`, `vilt`), we have something

```python
if not is_torch_greater_or_equal_than_xxx:
    logger.warning(
        f"You are using torch=={torch.__version__}, but torch>=xxx is required to use "
        "TapasModel. Please upgrade torch."
    )
```

However, I understand it's not really feasible to do what you describe in a modeling file, but it could be done in a processor file.

Happy to apply the suggestion for this single file `image_processing_pix2struct.py`.